### PR TITLE
Moving from travis to circleci

### PR DESCRIPTION
**Pull request by algojack**
* removed bionic 18 as it doesn't exist in circleci machine images (16 - xenial and 20 - focal left)
* indexer test is flaky, gives error 56 sometimes. Applied recommended circleci solution, and adjusted sleep per Will, but it doesn't completely fix it. Few restarts help.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        type: string
+    machine:
+      image: << parameters.machine >>
+    resource_class: medium
+    steps:
+      - checkout
+      - run:
+          name: Install docker
+          command: |
+            export SUDO=sudo
```
Comment (onetechnical) [inline .circleci/config.yml:36]: If you're just sudoing the rest, why not do it explicitly?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            $SUDO apt -y install docker-ce docker-ce-cli containerd.io
+      - run:
+          name: CircleCI docker download fix
+          command: sudo iptables -I INPUT -m conntrack --ctstate INVALID -j DROP
+      - run:
+          name: Stand up sandbox
+          command: |
+            export CONFIG=<< parameters.config >>
+            set -e
+            if [ -n "$DOCKER_PASSWORD" ]; then echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin ; fi
```
Comment (onetechnical) [inline .circleci/config.yml:52]: I think we can remove this line altogether, unless you're using a private repo.

**Review by onetechnical — CHANGES_REQUESTED** [review]
A couple questions. Love the matrix implementation. :)

File: .circleci/config.yml (hunk 2 above)
Comment (egieseke) [reply .circleci/config.yml:52]: I think that the login helps performance when using docker to pull images.

**Review by egieseke — APPROVED** [review]
Looks good.

File: .circleci/config.yml (hunk 1 above)
Comment (algojack) [reply .circleci/config.yml:36]: had to change too many times between having to do sudo, and having not to do sudo, so I just decided to put it up as a flag. I can remove if we want.

File: .circleci/config.yml (hunk 2 above)
Comment (algojack) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 1 above)
Comment (algojack) [reply .circleci/config.yml]: removed

**Review by onetechnical — APPROVED** [review]

File: sandbox [hunk 3]
```diff
@@ -174,8 +174,10 @@ function progress_bar {
 
 sandbox () {
   status_helper () {
+    sleep 10
     statusline "algod - goal node status"
     goal_helper node status
+    sleep 10
     statusline "indexer - health"
     curl -s "localhost:8980/health?pretty"
   }
```
Comment (winder) [inline sandbox:183]: you don't want to sleep here because it would mean `./sandbox status` would always take 20 seconds.

File: sandbox [hunk 4]
```diff
@@ -430,13 +432,13 @@ sandbox () {
         rebuild_if_needed >> "$SANDBOX_LOG" 2>&1               & spinner
         echo "* docker-compose up -d"   >> "$SANDBOX_LOG"
         dc up -d                        >> "$SANDBOX_LOG" 2>&1 & spinner
-        sleep 1                                                & spinner
+        sleep 10                                               & spinner
```
Comment (winder) [inline sandbox:435]: I guess this needs to be something in the 20-30 second range in order to succeed in your tests... that seems like too much just to get tests passing when end users are generally fine with 1 second.

I'm not sure the status is really needed though, how about leaving it at 1 second and removing the `status_helper` call just below these lines? That information might not be very useful so perhaps skip it to get tests running.

Comment (winder) [reply sandbox:435]: This isn't acceptable, it means that in normal operation most of the startup time is an artificial sleep. You're welcome to create a new test outside of the production script, but I need to revert this change.

**Pull request merged**